torch/jit/_isinstance.py

```diff
@@ -0,0 +1,109 @@
+from typing import List, Dict, Tuple, Union, Optional
+
+
+def get_origin(the_type):
+    return getattr(the_type, "__origin__", None)
+
+
+def get_args(the_type):
+    return getattr(the_type, "__args__", None)
+
+
+def check_args_exist(the_type):
+    if the_type is List or the_type is list:
+        raise RuntimeError(
+            "Attempted to use List without a "
+            "contained type. Please add a contained type, e.g. "
+            "List[int]"
+        )
+    elif the_type is Tuple or the_type is tuple:
+        raise RuntimeError(
+            "Attempted to use Tuple without a "
+            "contained type. Please add a contained type, e.g. "
+            "Tuple[int]"
+        )
+    elif the_type is Dict or the_type is dict:
+        raise RuntimeError(
+            "Attempted to use Dict without "
+            "contained types. Please add contained type, e.g. "
+            "Dict[int, int]"
+        )
+    elif the_type is None or the_type is Optional:
+        raise RuntimeError(
+            "Attempted to use Optional without a "
+            "contained type. Please add a contained type, e.g. "
+            "Optional[int]"
+        )
+
+
+def generics_checker(the_obj, the_type):
```

Comment:
Doesn't have to be in this diff.

Could you add a TODO for implementing check for "future"?